pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go

```diff
@@ -423,17 +451,127 @@ func TestWithBinding(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			state := framework.NewCycleState()
 			node := &framework.NodeInfo{}
 			node.SetNode(test.Node)
 			p := &VolumeZone{
 				pvLister,
 				pvcLister,
 				scLister,
 			}
-			gotStatus := p.Filter(context.Background(), nil, test.Pod, node)
-			if !reflect.DeepEqual(gotStatus, test.wantStatus) {
-				t.Errorf("status does not match: %v, want: %v", gotStatus, test.wantStatus)
+			_, gotPreFilterStatus := p.PreFilter(ctx, state, test.Pod)
+			if !gotPreFilterStatus.IsSuccess() {
+				if !reflect.DeepEqual(gotPreFilterStatus, test.wantStatus) {
+					t.Errorf("status does not match: %v, want: %v", gotPreFilterStatus, test.wantStatus)
+				}
+			} else {
+				gotStatus := p.Filter(ctx, state, test.Pod, node)
+				if !reflect.DeepEqual(gotStatus, test.wantStatus) {
+					t.Errorf("status does not match: %v, want: %v", gotStatus, test.wantStatus)
+				}
 			}
 		})
 	}
 }
+
+func BenchmarkVolumeZone(b *testing.B) {
+	tests := []struct {
+		Name       string
+		Pod        *v1.Pod
+		NumPV      int
+		NumPVC     int
+		NumNodes   int
+		wantStatus *framework.Status
+	}{
+		{
+			Name:     "zone label matched",
+			Pod:      createPodWithVolume("pod_0", "Vol_Stable_0", "PVC_Stable_0"),
+			NumPV:    10000,
+			NumPVC:   10000,
+			NumNodes: 1000,
+		},
+	}
+
+	for _, tt := range tests {
+		b.Run(tt.Name, func(b *testing.B) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			nodes := makeNodesWithTopologyZone(tt.NumNodes)
+			pl := newPluginWithListers(ctx, b, []*v1.Pod{tt.Pod}, nodes, makePVCsWithPV(1), makePVsWithZoneLabel(1))
+
+			p := pl.(*VolumeZone)
+			state := framework.NewCycleState()
+
+			b.ResetTimer()
+			_, gotPreFilterStatus := p.PreFilter(ctx, state, tt.Pod)
+			if !gotPreFilterStatus.IsSuccess() {
+				if !reflect.DeepEqual(gotPreFilterStatus, tt.wantStatus) {
```

Comment:
This will affect the benchmark. Since this is already tested in functional unit tests, you can just remove this check.